c++/src/kj/compat/http.h

```diff
@@ -52,10 +52,7 @@ namespace kj {
   MACRO(PURGE) \
   MACRO(OPTIONS) \
   MACRO(TRACE) \
-  /* standard methods */ \
-  /* */ \
-  /* (CONNECT is intentionally omitted since it should be handled specially in HttpServer) */ \
```

Comment:
It's kind of precarious to add CONNECT here because it means we have to be very careful that the application cannot use it. For example, `HttpClient::request()` really needs to throw if given `CONNECT` as the method, otherwise request smuggling vulnerabilities may open up. It's also dangerous if an HTTP server tries to process a `CONNECT` request as if it were a generic method.

I do see that this creates a problem for `HttpHeaders::tryParseRequest()`... we would want that to support parsing CONNECT requests, too. Maybe the solution is to extend the `RequestOrProtocolError` `OneOf` to include `ConnectRequest` as a completely separate type? That way no one can accidentally handle `CONNECT` the same as everything else.